### PR TITLE
[Refactor] Improve docker image build process and add some details

### PR DESCRIPTION
**Pull request by zgjja**
1. Delete the third party image source and use the NVIDIA's official image to build the runtime environment
2. Add detailed document
3. Add docker-compose support
4. Improve `.gitignore` and `.pre-commit-config.yaml`
5. Add cmake-format config

### CODE REVIEW

File: docker/README [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+* nvidia-container-toolkit
+* (Optional but **recommended**) docker-compose
+
+## Usage
+
+1. (With docker-compose) configure the `.env` file, change `DATA_DIR` to your mount point, such as your code or data folder, etc, comment the `volumes` in docker compose file if not necessariy needed
+
+2. Build image:
+```bash
+docker compomse -f docker-compose.yml build
```
Comment (wang-xinyu) [inline docker/README:20]: typo: compomse

File: docker/README [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# Tutorials
```
Comment (wang-xinyu) [inline docker/README:1]: The filename needs a suffix .md, use README.md.

File: docker/x86_64.dockerfile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+## override older cmake
+RUN find /usr/local/share -type d -name "cmake-*" -exec rm -rf {} + \
+&& curl -fsSL "https://github.com/Kitware/CMake/releases/download/v3.29.0/cmake-3.29.0-linux-x86_64.sh" \
+-o cmake.sh && bash cmake.sh --skip-license --exclude-subdir --prefix=/usr/local && rm cmake.sh
+
+## opencv contrib
+RUN git clone -b 4.x https://github.com/opencv/opencv_contrib.git
+
+## opencv
```
Comment (wang-xinyu) [inline docker/x86_64.dockerfile:22]: Is it possible to install pre-built opencv?  Building from src will be slow.

File: docker/README (hunk 1 above)
Comment (zgjja) [reply docker/README]: fixed

File: docker/README (hunk 2 above)
Comment (zgjja) [reply docker/README]: fixed

File: docker/x86_64.dockerfile (hunk 3 above)
Comment (zgjja) [reply docker/x86_64.dockerfile]: Add package installation as default method and leave the source build method commented for demo.

**Comment by wang-xinyu** [comment]
@zgjja LGTM. Thanks for the contribution!

**Pull request merged**